### DISCUSSION TITLE
Why the need for `python/get_python.sh`? Can I not use the system packages?

### DISCUSSION THREAD

**Discussion by Ashvith10** [o3de/o3de#16885] [Q&A]
I'm trying to add o3de to NixOS, and before writing an expression, I'm trying to see whether it is possible for me to build this using `nix-shell`. Now obviously, the Shell and Python script is failing which is explained [here](https://unix.stackexchange.com/a/751549/491279). I could try to patch them for  the time-being, but even if I do so, including this script would probably go against the reproducibility principles of Nix, so I was wondering if I could simply substitute the already available Python package available?

**Comment by nick-l-o3de** [comment]
If you're compiling yourself, and you can ensure the same python runtime gets embedded in the editor and other tools as you use on the CLI and from the environment, yes.  The main thing is that python is not binary ABI stable, meaning, if you link editor binary or other binary to one python library (python3.10.a) and then make it try to load a different one (python3.14.so) into the runtime, there will be offset and other memory issues even if you somehow force it to work.

But as long as you're compiling yourself and can ensure that the same python is used universally, it should be fine.
